src/wasm/wasm-binary.cpp

```diff
@@ -1913,6 +1911,17 @@ void WasmBinaryBuilder::visitSetGlobal(SetGlobal* curr) {
   auto index = getU32LEB();
   curr->name = getGlobalName(index);
   curr->value = popNonVoidExpression();
+  auto* global = wasm.getGlobalOrNull(curr->name);
+  if (global) {
+    curr->type = global->type;
+    return;
+  }
+  auto* import = wasm.getImportOrNull(curr->name);
+  if (import && import->kind == ExternalKind::Global) {
+    curr->type = import->globalType;
+    return;
+  }
+  throwError("bad set_global");
   curr->finalize();
```

Comment:
I may be missing something, but I don't think the code added to this function is needed - the type of `SetGlobal` is always `none`, as it does not return a value.